platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -57,3 +57,39 @@ include::sample[dir="snippets/bestPractices/usePluginsBlock/groovy/do-this",file
 
 === Tags
 `#structuring-builds`
+
+[[avoid_duplicate_dependencies]]
+== Avoid Declaring Dependencies Multiple Times
+
+You should avoid duplicating dependencies in your build scripts, whether with different versions or in different dependency configurations.
+
+=== Explanation
+
+Duplicating dependencies in build scripts can lead to several problems:
+
+* Conflicting versions may cause unexpected behavior or runtime errors due to version mismatches
+* Confusion during maintenance of build scripts and dependencies
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/avoidDuplicateDependencies/kotlin",files="build.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/avoidDuplicateDependencies/groovy",files="build.gradle[tags=avoid-this]"]
+====
+
+<1> Dependency on library version `1.10.0`
+<2> Redundant declaration that can be removed
```

Comment:
This isn't _redundant_, it's a request for a different version that gets replaced during resolution with the higher version.

This is actually an even _worse_ problem than _redundancy_.

Reply:
Not declaring 2 different versions?  Yes, probably?

Reply:
Ok, I'll make it as separate one. and think about the good example.